build/commands/lib/test.js

```diff
@@ -190,128 +207,172 @@ const runTests = (passthroughArgs, suite, buildConfig, options) => {
+          || options.manual_android_test_device,
+        'Only x86 and x64 builds can be run automatically. For other builds please run test device manually and specify manual_android_test_device flag.',
+      )
+
+      if (!options.manual_android_test_device) {
+        runArgs.push(
+          `--avd-config=tools/android/avd/proto/${options.android_test_emulator_name}.textpb`,
         )
+      }
+    }
 
-        if (!options.manual_android_test_device) {
-          runArgs.push(
-            `--avd-config=tools/android/avd/proto/${options.android_test_emulator_name}.textpb`,
-          )
-        }
+    if (config.isTeamcity) {
+      // Stdout and stderr must be separate for a test launcher.
+      runOptions.stdio = 'inherit'
+    }
+
+    let progStatus = 0
+
+    if (config.isIOS()) {
+      const outputDir = path.join(config.outputDir, `run_${testSuite}_out`)
+      if (fs.existsSync(outputDir)) {
+        fs.rmSync(outputDir, { recursive: true })
+      }
+      fs.mkdirSync(path.join(outputDir, 'iossim'), { recursive: true })
+      fs.mkdirSync(path.join(outputDir, 'output'), { recursive: true })
+
+      let xcodeBuildVersion = options.ios_xcode_build_version
+      if (xcodeBuildVersion == null) {
+        xcodeBuildVersion = util
+          .run('xcodebuild', ['-version'])
+          .stdout.toString()
+          .trim()
+          .split(' ')
+          .pop()
       }
 
-      if (config.isTeamcity) {
-        // Stdout and stderr must be separate for a test launcher.
-        runOptions.stdio = 'inherit'
+      runArgs.push('--app', getTestBinary(Config, testSuite))
+      runArgs.push('--runtime-cache-prefix', `${outputDir}/Runtime-ios-`)
+      runArgs.push('--iossim', `${outputDir}/iossim`)
+      runArgs.push('--out-dir', `${outputDir}/output`)
+      runArgs.push('--xctest')
+      runArgs.push('--xcode-build-version', xcodeBuildVersion)
+      if (config.targetEnvironment === 'simulator') {
+        runArgs.push(
+          '--platform',
+          options.ios_simulator_platform || 'iPhone 16',
+        ) // update as needed for version
+        runArgs.push('--version', options.ios_simulator_version || '18.4') // should match ios_deployment_target
+      } else if (config.targetEnvironment === 'device') {
+        // TODO(bridiver)
+        // runArgs.push('--xcodebuild-device-runner')
       }
+      let iosRunOptions = Object.assign({}, runOptions)
+      // run.py doesn't like it when you mess with PYTHONPATH
+      delete iosRunOptions.env.PYTHONPATH
+      progStatus = util.run(
+        path.join(config.srcDir, 'ios/build/bots/scripts/run.py'),
+        runArgs,
+        iosRunOptions,
+      ).status
 
-      let prog = util.run(getTestBinary(testSuite), runArgs, runOptions)
+      if (options.output_xml) {
+        util.run(
+          'python3',
+          [
+            path.join('tools', 'test', 'chromium_to_junit_converter.py'),
+            path.join(outputDir, 'output', 'full_results.json'),
```

Comment:
Failed tests do not display anything other than `Test crash (expected: PASS)`. I believe you should process `summary.json` to display actually usable information. See: https://ci.brave.com/job/brave-core-build-pr-ios/job/PR-29992/49/execution/node/55/ws/src/out/ios_Debug_simulator/run_brave_components_unittests_out/output/summary.json/*view*/